src/Symfony/Component/Security/Guard/Token/PostAuthenticationGuardToken.php

```diff
@@ -28,9 +27,9 @@ class PostAuthenticationGuardToken extends AbstractToken implements GuardTokenIn
     private $providerKey;
 
     /**
-     * @param UserInterface   $user        The user!
-     * @param string          $providerKey The provider (firewall) key
-     * @param (Role|string)[] $roles       An array of roles
+     * @param UserInterface $user        The user!
+     * @param string        $providerKey The provider (firewall) key
+     * @param string[]      $roles       An array of roles
```

Comment:
Although the role is deprecated the `$roles` argument can still accept `Roles` object..

this change breaks my phpstan when I have something like this:

`        $newToken = new PostAuthenticationGuardToken($user, 'secured_area', $user->getRoles());`

would it be acceptable to revert this docblock change?

Reply:
We normally do not document the deprecated arguments anymore. Same as if we deprecate an argument and use func_get_arg for BC. Otherwise people could think the deprecated way is still the right way and write code accordingly. So IMO we should not change that. Either fix the deprecation or ignore the phpstan error.

Reply:
> We normally do not document the deprecated arguments anymore. 

I guess the same does not apply [here](https://github.com/symfony/symfony/blob/4.4/src/Symfony/Component/Security/Core/User/UserInterface.php#L49) because it is a return argument?

Reply:
It should be changed there as well.